### PR TITLE
[VBLOCKS-3510] Allow for disabling of built-in firebase service

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

This PR adds APIs to allow users to disable the built in firebase service.

## Breakdown

- Add opt-out configuration and feature to the SDK.
- Add documentation for this feature and its use-cases.

## Validation

- Manually tested locally with a 3rd party firebase messaging library.

## Additional Notes

N/A

### CODE REVIEW

File: android/src/main/java/com/twiliovoicereactnative/VoiceApplicationProxy.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ public void onServiceDisconnected(ComponentName name) {
       voiceServiceApi = null;
```
Comment (afalls-twilio) [inline android/src/main/java/com/twiliovoicereactnative/VoiceApplicationProxy.java:35]: The only change in this file is additional new lines, could we just revert the changes?

**Review by afalls-twilio — CHANGES_REQUESTED** [review]
One little minor nit...

File: src/Voice.tsx [hunk 2]
```diff
@@ -578,6 +578,38 @@ export class Voice extends EventEmitter {
...
+  /**
+   * Handle Firebase messages from an out-of-band Firebase messaging service.
+   *
+   * @remarks
+   *
+   * Note that this method only works on Android platforms, and will only work
+   * when the built-in Firebase messaging service as been opted-out.
+   *
+   * Unsupported platforms:
+   * - iOS
```
Comment (kpchoy) [inline src/Voice.tsx:590]: Do you want to add a note if this will supported in the future, or if we do not plan to support in the future?

Comment (mhuynh5757) [reply src/Voice.tsx:590]: It won't be necessary to support in the future, as iOS does not have the same Firebase messaging service limitations as Android does.

File: android/src/main/java/com/twiliovoicereactnative/VoiceApplicationProxy.java (hunk 1 above)
Comment (afalls-twilio) [reply android/src/main/java/com/twiliovoicereactnative/VoiceApplicationProxy.java]: ahh no, just nonsense.. not sure how they got introduced.. thanks!

**Review by afalls-twilio — APPROVED** [review]
LGTM

**Review by kpchoy — APPROVED** [review]
